src/DS/rds.c

```diff
@@ -699,8 +705,13 @@ static int ds_rds_create_from_dom(xmlDocPtr* ret, xmlDocPtr sds_doc, xmlDocPtr t
 
 	xmlDOMWrapCtxtPtr sds_wrap_ctxt = xmlDOMWrapNewCtxt();
 	xmlNodePtr sds_res_node = NULL;
-	xmlDOMWrapCloneNode(sds_wrap_ctxt, sds_doc, xmlDocGetRootElement(sds_doc),
-			&sds_res_node, doc, NULL, 1, 0);
+	if (clone) {
+		xmlDOMWrapCloneNode(sds_wrap_ctxt, sds_doc, xmlDocGetRootElement(sds_doc),
+				&sds_res_node, doc, NULL, 1, 0);
+	} else {
+		sds_res_node = xmlDocGetRootElement(sds_doc);
+		xmlDOMWrapAdoptNode(sds_wrap_ctxt, sds_doc, sds_res_node, doc, NULL, 0);
+	}
```

Comment:
`ds_rds_create_from_dom_clone` × `ds_rds_create_from_dom` pair is definitely more readable compared to some cryptic boolean as the last argument.

Reply:
Great! Indeed, having 2 functions will be more readable. But then, how to avoid code duplication? `ds_rds_create_from_dom` is about 130 lines, I don't wish to copy them.